File: app/src/main/java/com/babic/filip/movieshack/database/DatabaseInterface.java
package com.babic.filip.movieshack.database;

import com.babic.filip.movieshack.model.Movie;

import java.util.List;

public interface DatabaseInterface {

    List<Movie> getMoviesByType(String type);

    void addMovies(List<Movie> movies);

    void updateMovie(Movie movie);

    Movie getMovieById(String id);

    void clearMoviesByType(String type);
}

File: app/src/main/java/com/babic/filip/movieshack/database/DatabaseInterface.kt
package com.babic.filip.movieshack.database

import com.babic.filip.movieshack.model.Movie

interface DatabaseInterface {

    fun getMoviesByType(type: String): List<Movie>

    fun addMovies(movies: List<Movie>)

    fun updateMovie(movie: Movie)

    fun getMovieById(id: String): Movie

    fun clearMoviesByType(type: String)
}
